lib/reply.js

```diff
@@ -103,6 +103,15 @@ Reply.prototype.redirect = function (code, url) {
   this.header('Location', url).code(code).send()
 }
 
+Reply.prototype.notFound = function () {
+  if (this.context._fastify === null) {
+    throw new Error('Cannot call "reply.notFound()" inside a 404 handler')
```

Comment:
Could you add a test for this if?

Reply:
I'm not sure that throwing is the right thing to do.
Maybe log a warn and send to the user a basic 404 error could be the right choice.